### ISSUE TITLE
imported classes not being recognized on cljc reader conditional

### ISSUE THREAD

**Issue by ericdallo** [clj-kondo/clj-kondo#1363]
**version**

2021.08.06

**platform**

JVM

**editor**

clj-kondo cli but happens with clojure-lsp as well

**problem**

It seems clj-kondo doesn't understand properly imports inside conditional reader macros, the analysis output seems perfect, but the findings not.

**repro**

`foo.cljc`
```clojure
(ns foo
  (:require
   #?(:clj [clojure.string :refer [split-lines]])))

(split-lines "")
```

`clj -Sdeps '{:deps {clj-kondo/clj-kondo {:mvn/version "2021.08.06"}}}' -m clj-kondo.main --lint foo.cljc`
```bash
foo.cljc:6:2: error: Unresolved symbol: split-lines                                                                                                                                           
linting took 101ms, errors: 1, warnings: 0 
```

**expected behavior**

No unresolved symbol

**Comment by ericdallo** [comment]
Oh just found this however works:
```
(ns clojure-sample.foo
  (:require
   #?(:clj [clojure.string :refer [split-lines]])))


#?(:clj
   (split-lines ""))

```

**Comment by ericdallo** [comment]
Sorry, it seems to make sense add into the clj reader conditional indeed
